internal/terraform/node_action_trigger_plan.go

```diff
@@ -100,6 +101,21 @@ func (n *nodeActionTriggerPlanExpand) DynamicExpand(ctx EvalContext) (*Graph, tf
 		for _, key := range keys {
 			absResourceInstanceAddr := n.lifecycleActionTrigger.resourceAddress.Absolute(module).Instance(key)
 
+			if n.resourceTargets != nil {
```

Comment:
Why is this necessary, is this another quirk of actions runtime behaviors? The targets transformer should drop all unnecessary nodes from the graph, so I would hope we wouldn't need to manage that behavior here as well as in the transformer?